src/Native/Unix/System.Native/pal_io.c

```diff
@@ -121,8 +121,8 @@ c_static_assert(PAL_SEEK_CUR == SEEK_CUR);
 c_static_assert(PAL_SEEK_END == SEEK_END);
 
 // Validate our PollFlags enum values are correct for the platform
-// HACK: AIX values are different; we convert them between PAL_POLL and POLL now
-#ifndef _AIX
+// HACK: AIX and Haiku values are different; we convert them between PAL_POLL and POLL now
+#if !(defined (_AIX) || defined (__HAIKU__))
```

Comment:
We're now always translating on all platforms, even if the values are the same, right? If so, we can just remove these c_static_asserts entirely.  They're there only in cases where we're relying on the PAL values matching the platform.